### PR TITLE
DO NOT MERGE: Add PyHeron packaging script

### PR DESCRIPTION
**Pull request by taishi8117**
This is a working branch.

Still a prototype and not ready to be merged.

Depends on #1198

### CODE REVIEW

File: heron/common/src/python/BUILD [hunk 1]
```diff
@@ -29,3 +29,10 @@ pex_library(
   ]
 )
 
+# This is common package for pyheron
+pex_library(
+  name = 'pyheron_common',
```
Comment (kramasamy) [inline heron/common/src/python/BUILD]: please change the name to ```common_py```

File: heron/streamparse/src/python/BUILD [hunk 2]
```diff
@@ -6,8 +6,17 @@ pex_library(
...
-  reqs = ['colorlog==2.6.1', 'pyyaml==3.10'],
 )
+
+# for pyheron egg production
+pex_binary(
+  name = "pyheron_pkg",
+  srcs = glob(["**/*.py"]),
+  deps = [
+    ":pyheron_lib",
+    "//heron/examples/src/python:pyheron_example_pkg"
```
Comment (kramasamy) [inline heron/streamparse/src/python/BUILD]: it should be heron-examples-python?

File: scripts/packages/pyheron/setup.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+setup(
+  name='pyheron',
+  version='0.0.1',
+  description='Python API for Heron topology',
+  long_description='Long description about pyheron',
+
+  url='https://github.com/twitter/heron',
+
+  author='Heron Release Team',
+  author_email='streaming-compute@twitter.com',
```
Comment (kramasamy) [inline scripts/packages/pyheron/setup.py]: please change it heronstreaming@gmail.com

File: scripts/packages/pyheron/setup.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+here = os.path.abspath(os.path.dirname(__file__))
+requirements = ['colorlog==2.6.1',
+                'protobuf==2.5.0',
+                'setuptools==18.0.1']
+
+setup(
+  name='pyheron',
+  version='0.0.1',
+  description='Python API for Heron topology',
+  long_description='Long description about pyheron',
```
Comment (kramasamy) [inline scripts/packages/pyheron/setup.py]: Pyheron package allows a developer to write python topology in Heron. Pyheron is backward compatible with the popular python API called streamparse. It can be run in various clusters including Mesos/Aurora, Mesos/Marathon, YARN, etc.

**Comment by taishi8117** [comment]
Also, we need to create a PyPI account.

**Comment by objmagic** [comment]
#1198 is merged. Merging with master and restarting CI now.

File: scripts/packages/pyheron/setup.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    'Development Status :: 3 - Alpha',
+
+    'Intended Audience :: Developers',
+    'Topic :: Realtime streaming',
+
+    'Programming Language :: Python :: 2.7',
+  ],
+
+  keywords='heron topology python',
+  packages=find_packages(),
```
Comment (objmagic) [inline scripts/packages/pyheron/setup.py]: we should specify license somewhere? Is this related to #1259 ?
image

**Comment by taishi8117** [comment]
We will close this PR, as @kramasamy will push the script integrated with Bazel.
